dts/msm8974/msm8974-fairphone-fp2.dts

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	model = "Fairphone 2";
+	compatible = "fairphone,fp2", "qcom,msm8974", "lk2nd,device";
+	qcom,msm-id = <217 0x10000>, <218 0x10000>;
+	qcom,board-id = <8 0>;
```

Comment:
This looks like [`msm8974pro-ab-pm8941-mtp.dts`](https://github.com/msm8916-mainline/android_kernel_qcom_msm8916/blob/master/arch/arm/boot/dts/qcom/msm8974pro-ac-pm8941-mtp.dts). Can you name it that way and if there is something on the bootloader cmdline that would allow identifying the fp2? There might be other devices using the same `qcom,board-id`.

If you can't find anything another popular option would be matching the panel names (`lk2nd,match-panel`) after you have listed them similar to other devices. This would also enable using it for panel selection in the mainline Linux kernel (iirc fp2 also has multiple possible panels).

Also: Can you check that you can also boot downstream with this? E.g. TWRP or LineageOS.

Reply:
Booting TWRP does work from lk2nd.

Example cmdline in TWRP:
```
console=ttyHSL0,115200,n8 androidboot.hardware=qcom user_debug=31 msm_rtb.filter=0x3b7 ehci-hcd.park=3 androidboot.bootdevice=msm_sdcc.1 lk2nd androidboot.emmc=true androidboot.serialno=eb1ef6b5 androidboot.baseband=msm mdss_mdp.panel=1:dsi:0:qcom,mdss_dsi_otm1902b_1080p_cmd
```

The panel on FP2 is either `mdss_mdp.panel=1:dsi:0:qcom,mdss_dsi_otm1902b_1080p_cmd` or `mdss_mdp.panel=1:dsi:0:qcom,mdss_dsi_s6d6fa1_1080p_video` which is unique enough I think.

Reply:
So I will add similar to this https://github.com/msm8916-mainline/lk2nd/commit/be73ef2bc8bd7ef2fb8207f319ec85757b06cf0f, yes? These compatibles are currently not used in the mainline-ish repo but support for the panel is kind of wonky in terms of mainline support (incl. the touchscreen used on that one) and I've been using only the s6d6fa1 one in the last years. I hope lk2nd won't fail if it can't find the panel nodes in dts?

```
lk2nd,match-panel;

panel {
    compatible = "fairphone,fp2-panel";

    qcom,mdss_dsi_otm1902b_1080p_cmd {
        compatible = "fairphone,fp2-panel-otm1902b";
    };
    qcom,mdss_dsi_s6d6fa1_1080p_video {
        compatible = "fairphone,fp2-panel-s6d6fa1";
    };
}
```

Reply:
> I hope lk2nd won't fail if it can't find the panel nodes in dts?

It shouldn't. Looks fine to me

Reply:
Added. Boots fine with both panels.

![IMG_20220127_202937](https://user-images.githubusercontent.com/3768500/151430123-78715bf4-174b-4bd2-89b2-62ec8a44ced5.jpg)

Reply:
Sorry if I wasn't clear enough, but what we need is a generic `msm8974pro-ab-pm8941-mtp.dts` file with a `fairphone-fp2` subnode, similar to [`msm8916-mtp.dts`](https://github.com/msm8916-mainline/lk2nd/blob/master/dts/msm8916/msm8916-mtp.dts) and [`msm8916-mtp-smb1360.dts`](https://github.com/msm8916-mainline/lk2nd/blob/master/dts/msm8916/msm8916-mtp-smb1360.dts) etc. The fp2 should not be on top-level. :)

Reply:
Updated, is this what you had in mind?

Reply:
Yep. :)